lib/timetable-form/Form.tsx

```diff
@@ -1,38 +1,63 @@
 import { useState } from "react";
 import styles from "./styles.module.css";
-
-const stageNames = [
-  "Timetable published",
-  "Draft plan for public consultation published",
-  "Public consultation start",
-  "Public consultation end",
-  "Submit plan for examination",
-  "Examination hearing start",
-  "Planning inspectorate examination start",
-  "Planning inspectorate examination end",
-  "Planning inspectorate found sound",
-  "Examination hearing end",
-  "Inspector report published",
-  "Plan adopted",
-] as const;
-
-type StageName = (typeof stageNames)[number];
+import { DevelopmentPlanTimetable, Stages } from "../types/timetable";
+import { stageNames } from "../constants";
+import { randomUUID } from "crypto";
 
 const defaultDate = "";
 
-type Stages = {
-  [key in StageName]: string;
-};
-
 const initialStages = Object.fromEntries(
   stageNames.map((stage) => [stage, defaultDate])
 ) as Stages;
 
+type FormData = {
+  LPA: string;
+  stages: Stages;
+};
+
+const formStateToDevelopmentPlanTimetables = (
+  state: FormData
+): DevelopmentPlanTimetable[] => {
+  return Object.entries(state.stages).map(([event, eventDate]) => ({
+    reference: randomUUID(),
+    name: "",
+    developmentPlan: "dorcester-new-local-plan", //this will be coming from a dropdown on the form
```

Comment:
BTW, I don't know that it _will_ be coming from a dropdown (we haven't figured out how this is going to work yet)